### PR TITLE
SelectDropdown: fix visual issues

### PR DESCRIPTION
**Pull request by retrofox**
This PR fix some visual issues of the `<SelecDropdown />` component. Please take a look to #4971 to read more about these ones.

### Issues

#4971 

### Screenshots

**devdocs testing page**
http://calypso.localhost:3000/devdocs/design/select-dropdown
select-dropdown-good-08

---------------------------
**posts count - english**
select-dropdown-good-05

---------------------------
**posts count - spanish**
select-dropdown-good-04

---------------------------
**posts count - no idea**
select-dropdown-good-06

cc @lancewillett @mtias

### CODE REVIEW

File: client/components/select-dropdown/docs/example.jsx [hunk 1]
```diff
@@ -43,10 +43,10 @@ var SelectDropdownDemo = React.createClass( {
 	},
 
 	render: function() {
-		var toggleButtonsText = this.state.compactButtons ? 'Normal Buttons' : 'Compact Buttons';
+		let toggleButtonsText = this.state.compactButtons ? 'Normal Buttons' : 'Compact Buttons';
```
Comment (gwwar) [inline client/components/select-dropdown/docs/example.jsx]: This can be `const`

File: client/components/select-dropdown/index.jsx [hunk 2]
```diff
@@ -24,13 +24,39 @@ import Count from 'components/count';
  * Module variables
  */
 const { Component, PropTypes } = React;
-var noop = () => {};
+const noop = () => {};
```
Comment (gwwar) [inline client/components/select-dropdown/index.jsx]: I prefer to either do `import { noop } from 'lodash'` or use the anon function directly.

**Comment by gwwar** [comment]
It looks like your `FeaturesList` removal is also in this branch. Do you mind removing that so the PR is a little smaller to look at?

**Comment by gwwar** [comment]
I'm seeing some visual issues with the compact version:
screen shot 2016-08-11 at 11 37 27 am

File: client/components/select-dropdown/docs/example.jsx (hunk 1 above)
Comment (retrofox) [inline client/components/select-dropdown/docs/example.jsx]: I wonder why I don't get the eslint warning

image

**Comment by retrofox** [comment]
@gwwar thanks for the review. updated & rebased.

**Comment by retrofox** [comment]
btw I forgot tell you I've updated the screenshot at the PR description.

File: client/components/select-dropdown/docs/example.jsx [hunk 3]
```diff
@@ -43,10 +43,12 @@ var SelectDropdownDemo = React.createClass( {
...
 
 	render: function() {
-		var toggleButtonsText = this.state.compactButtons ? 'Normal Buttons' : 'Compact Buttons';
+		const toggleButtonsText = this.state.compactButtons
+			? 'Normal Buttons'
+			: 'Compact Buttons';
 
 		return (
-			<div className="design-assets__group">
+			<div className="design-assets__group" style={ { height: 500 } }>
```
Comment (gwwar) [inline client/components/select-dropdown/docs/example.jsx]: Not a huge deal since this is an example, but could we try to avoid an inline style here?

**Comment by gwwar** [comment]
👍 Almost there. Edge needs a few tweaks. We also need to retune our breakpoints with the draft drawer, but that should be done in another PR.

Edge ( Looks a bit off here )
screen shot 2016-08-11 at 3 36 47 pm
IE11 looks OK
screen shot 2016-08-11 at 3 53 03 pm

**Comment by retrofox** [comment]
> We also need to retune our breakpoints with the draft drawer

what do you mean?

**Comment by gwwar** [comment]
Try resizing the posts page, the controls will overlap with each other for a while before hitting existing media breakpoints.

**Comment by gwwar** [comment]
This is almost ready to go. Let's either fix the lowercase "published" text in Edge, or write up an issue to fix.

**Comment by rralian** [comment]
I'm not sure this is the right approach. In fairness I haven't been able to recreate the original issue, but going by the description it seems that the issue is the select-dropdown has a width according to the _selected item_, rather than expanding to contain the width of the widest item when it's open. Applying a min-width seems like a band-aid that won't address the issue well with i18n or in a general sense outside of the posts/pages page.

I think it would be better to apply a `white-space: nowrap` to the individual items in the dropdown and let the container expand to contain the items as necessary.

**Comment by retrofox** [comment]
it seems very reasonable @rralian. Testing your approach now.

**Comment by gwwar** [comment]
Without a maximum width, we'll likely run into issues trying to tune the controls in a given breakpoint (think of the varying widths we get from i18n). I think an ellipsis or fade for the label would be a better fit.

**Comment by retrofox** [comment]
select-dropdown

cc @rralian

**Comment by rralian** [comment]
What are you showing me here? Are you demonstrating that you can recreate the problem? Or did you try the white-space approach and are showing that there is still an issue?

**Comment by retrofox** [comment]
> In fairness I haven't been able to recreate the original issue, ...

showing how to recreate the original issue.

**Comment by gwwar** [comment]
On second thought after testing this in a few languages, our current labels aren't that long. We might run into a bit of control overlap, but it looks relatively minor.

**Comment by gwwar** [comment]
👍 Retested this and it works well. We may have a few minor issues with overlapping controls, but we can deal with that in additional PRs. Let's wait for a 👍 from @rralian before 🚢

**Comment by rralian** [comment]
> Let's wait for a 👍 from @rralian before 🚢

Well, like I said earlier I don't think this is the right approach. @retrofox said he was testing my suggestion, but I don't see a commit or mention of that since. I wouldn't merge this approach. We're applying a width that fixes the issue in a specific instance and applying it to the general component. That leaves us open to issues in other instances of the select dropdown or even in this instance in other i18n translations. Where really, the thing that we actually care about is that we don't want a break in the items.

> Without a maximum width, we'll likely run into issues trying to tune the controls in a given breakpoint (think of the varying widths we get from i18n). I think an ellipsis or fade for the label would be a better fit.

I think this is more of a corner case, but if we're concerned that i18n will cause issues, we could require a max-width to be passed into the component as a prop, forcing the implementer to consider i18n variance. I think that's a different issue and should be handled separately from this PR however.

**Comment by gwwar** [comment]
> I think this is more of a corner case, but if we're concerned that i18n will cause issues

I tested a few languages yesterday, and I don't think its a major issue. We can address this in a future PR if a dropdown is used with excessively long labels.

**Comment by retrofox** [comment]
Ok guys. I've updated this PR changing the way that the dropdown is built in terms of its size, position, etc. Tested under FF, Chrome and Chrome. No tested yet in IE.

Now if we not define the width of the component it will be adjusted in function of the max width value among its options elements. Take a look to some gifs at the PR description (update).

Now if we wanna fix a min-width you should override some css rules (like usual):

select-dropdown-good-09

**Comment by rralian** [comment]
The vertical alignment of the numbers seems off to me. I think the numbers are a little higher now. Here's an example from master, showing the number vertically centered in the "drafts" highlighted item.

screen shot 2016-08-18 at 2 11 37 pm

And here's an example from this branch, with the number a little too high.

screen shot 2016-08-18 at 2 12 28 pm

I think it's actually that this PR adds a few pixels to the bottom of the item. I notice that they're taller. To the point that the section-nav example has the last item cut-off quite a bit. This is due to that example applying a maximum height, but it demonstrates the issue.

screen shot 2016-08-18 at 2 53 32 pm

Last, I notice that this forces the entire select-dropdown to be wider, which can make it overlap other elements in section-nav. So we may need to adjust our responsive breakpoints to adjust. Here's an example in this branch:

screen shot 2016-08-18 at 3 13 27 pm

That same example in wpcalypso (ie master) also has some overlap, though it's less pronounced.

screen shot 2016-08-18 at 3 13 52 pm

File: client/components/select-dropdown/docs/example.jsx [hunk 4]
```diff
@@ -43,10 +43,12 @@ var SelectDropdownDemo = React.createClass( {
...
 
 	render: function() {
-		var toggleButtonsText = this.state.compactButtons ? 'Normal Buttons' : 'Compact Buttons';
+		const toggleButtonsText = this.state.compactButtons
+			? 'Normal Buttons'
+			: 'Compact Buttons';
 
 		return (
-			<div className="design-assets__group">
+			<div className="design-assets__group design-assets__group__select-dropdown">
```
Comment (mtias) [inline client/components/select-dropdown/docs/example.jsx]: Should be using `docs__` classes.

**Comment by retrofox** [comment]
- [x] The vertical alignment of the numbers seems off

image

- [x] I think it's actually that this PR adds a few pixels to the bottom of the item.

**current**
image

**before to fix**
image

**fixed**
image

> Last, I notice that this forces the entire select-dropdown to be wider, ...

It could happen because the dropdown always will be fixed to the max width of one of its options. This is its natural behavior now when the width isn't set explicitly. however I've optimized this adjustment to minimum possible.

> ... which can make it overlap other elements in section-nav. So we may need to adjust our responsive breakpoints to adjust.

This issue occurs in prod right now:

image

I suggest create an issue for this and fix it in another PR.

**Pull request merged**